examples/text/amazonreviewpolarity.py

```diff
@@ -31,19 +34,43 @@
 def AmazonReviewPolarity(root, split):
     """Demonstrating caching, extraction and sanity check pipelines."""
 
+    # Wrapping the URL into a IterDataPipe
     url_dp = IterableWrapper([URL])
+
+    # `.on_disk_cache` is the functional form of `OnDiskCacheHolder`, which caches the results from the
+    # subsequent DataPipe operations (until `.end_caching`) onto the disk to the path as specified by `filepath_fn`.
+    # In addition, since the optional argument `hash_dict` is given, the DataPipe will also check the hashes of
+    # the files before saving them. `.on_disk_cache` merely indicates that caching will take place,
+    # but the content of the DataPipe is unchanged. Therefore, `cache_compressed_dp` still contains a list of URL.
     cache_compressed_dp = url_dp.on_disk_cache(
         filepath_fn=lambda x: os.path.join(root, _PATH), hash_dict={os.path.join(root, _PATH): MD5}, hash_type="md5"
     )
+
+    # `GDriveReader` takes in URLs to GDrives files, and yields a tuple of file name and IO stream.
+    # After that, `.end_caching` saves the previous DataPipe's outputs onto the disk. In this case,
+    # the results from GDriveReader (i.e. the downloaded compressed archive) will be saved onto the disk.
+    # Upon saving the results, the DataPipe returns the paths to the cached files.
     cache_compressed_dp = GDriveReader(cache_compressed_dp).end_caching(mode="wb", same_filepath_fn=True)
```

Comment:
Super nit: We can split this line into two parts and comment on each op.
cache_compressed_dp = GDriveReader(cache_compressed_dp)
cache_compressed_dp = cache_compressed_dp.end_caching(mode="wb", same_filepath_fn=True)